scripts/file_functions.R

```diff
@@ -19,24 +19,36 @@ set_up_paths <- function(params) {
     return(paths)
 }
 
-set_up_paths_2 <- function(paths, params, facets){
+set_up_paths_2 <- function(paths, params, facets, display_facets){
   if (is.na(params$group_facet) || is.null(params$group_facet)) {
     paths$DEG_output <- file.path(paths$results, "DEG_lists")
-    paths$pathway_analysis <- file.path(paths$results, "pathway_analysis")
     paths$biosets_output <- paths$BMD_output
   } else {
     # make multiple outputs for different facets
     for(f in facets){
       paths$DEG_output[[f]] <- file.path(paths$results, "DEG_lists", paste0(f))
-      paths$pathway_analysis[[f]] <- file.path(paths$results, "pathway_analysis", paste0(f))
       paths$biosets_output[[f]] <- file.path(paths$BMD_output, paste0(f))
     }
   }
   lapply(paths$DEG_output, function(x) if(!dir.exists(x)) dir.create(x, recursive = TRUE))
-  lapply(paths$pathway_analysis, function(x) if(!dir.exists(x)) dir.create(x, recursive = TRUE))
   lapply(paths$biosets_output, function(x) if(!dir.exists(x)) dir.create(x, recursive = TRUE))
   return(paths)
 }
+
+set_up_paths_3 <- function(paths,params,display_facets){
```

Comment:
Just to make sure I understand: does this function replace set_up_paths_2?

Reply:
No, it's a third one, unfortunately. Because the GO analysis produces some text output, I need to be able to set the paths appropriately for that one after the `display_facets` have been set up. `set_up_paths_2()` needs the DEseq facets.

Reply:
It's not great because we have analysis (the GO output) that's based on the `display_facets` which should be just for visualization. But the GO report does some analysis too.